ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5839,13 +5842,33 @@ public HiveMetaHook getHook(
       }
     };
 
+    if ("rest".equals(conf.get("metastore.type", "hms"))) {
+      return getHiveIcebergRESTCatalog(conf, hookLoader);
+    }
+
     if (conf.getBoolVar(ConfVars.METASTORE_FASTPATH)) {
       return new SessionHiveMetaStoreClient(conf, hookLoader, allowEmbedded);
     } else {
       return RetryingMetaStoreClient.getProxy(conf, hookLoader, metaCallTimeMap,
           SessionHiveMetaStoreClient.class.getName(), allowEmbedded);
     }
   }
+  private IMetaStoreClient getHiveIcebergRESTCatalog(HiveConf conf, HiveMetaHookLoader hookLoader) throws MetaException {
```

Comment:
can we create IMetaStoreClientFactory and extract this logic there

Reply:
Refactored.